packages/calcite-components/src/components/button/button.scss

```diff
@@ -1,4 +1,29 @@
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-button-background-color: Specifies the background color of the component.
+ * @prop --calcite-button-text-color: Specifies the text and icon color of the component.
+ * @prop --calcite-button-box-shadow: Specifies the border color of the component.
+ * @prop --calcite-button-border-color: Specifies all border colors of the component.
+ * @prop --calcite-button-border-top-color: Specifies the top border color of the component.
+ * @prop --calcite-button-border-right-color: Specifies the right border color of the component.
+ * @prop --calcite-button-border-bottom-color: Specifies the bottom border color of the component.
+ * @prop --calcite-button-border-left-color: Specifies the left border color of the component.
+ *
+ */
+
 :host {
+  --calcite-button-background-color: var(--calcite-color-background);
+  --calcite-button-text-color: var(--calcite-color-text-1);
+  --calcite-button-box-shadow: var(--calcite-shadow-none);
+  --calcite-button-border-color: var(--calcite-color-transparent);
+  --calcite-button-border-top-color: var(--calcite-color-transparent);
+  --calcite-button-border-right-color: var(--calcite-color-transparent);
+  --calcite-button-border-bottom-color: var(--calcite-color-transparent);
+  --calcite-button-border-left-color: var(--calcite-color-transparent);
+
```

Comment:
Since there are so many places where these component variables need to be set based on prop combinations, it would be much more readable and discoverable if instead of assigning and reassigning all of them in the existing selector blocks below, if you could move all the css variable assignments up to the top of the file here, assigning them based on their prop combinations in a new set of `:host` css selectors that mirror their current placement in the rest of the css file.  That way you can see at a glance which prop combinations result in which default component variable settings.  You can then use assign those variables in the appropriate places below.

For example:
```
:host([kind="brand"]) {
    --calcite-button-text-color: var(--calcite-color-text-inverse);
    --calcite-button-background-color: var(--calcite-color-brand);
}
:host([kind="danger"]) {
    --calcite-button-text-color: var(--calcite-color-text-inverse);
    --calcite-button-background-color: var(--calcite-color-status-danger);
}
// additional kinds here and other prop combinations...

// And then you only have to set these values once with a rule like this that sits below all the prop-based assignments:
:host {
    button, a {
        color: var(--calcite-button-text-color);
        background-color: var(--calcite-button-background-color);
    }
}
```

Reply:
So I made some changes based on the suggestions you gave, the only one I didn't do was moving some things up to the top, I kept them relatively in the same positions because I didn't want to have to deal with potentially unwanted issues from repositioning some styles above others. I think the changes already went above the main focus of this effort which was just to swap out some values for tokens based on other PRs I saw. 

Let me know what you think about the changes and if they are close to what you had in mind.

Reply:
My only question remaining is the need for explicit border sides. Can we just do one? I think this aligns with other component efforts unless there is another reason @DitwanP

Reply:
I got rid of the border sides so the code referenced in this comment is now outdated there's just one again. @macandcheese